### ISSUE TITLE
Fix Q37 in benchmark suite

### ISSUE THREAD

**Issue by ray6080** [kuzudb/kuzu#2150]
Q37 has failed for a while in the benchmark suite. Either the ground truth is incorrect, or we have bugs for this query.
```
-NAME q37
-COMPARE_RESULT 1
-QUERY MATCH (a:Person)-[r:knows* SHORTEST 1..30]->(b:Person) WHERE a.ID = 94 RETURN COUNT(*)
---- 1
407396

```

**Comment by anuchak** [comment]
Yes this seems to be an issue for Shortest path queries, if I return path then `407396` seems to be the answer but if I return only the path length / do `COUNT(*)` then `407595` is returned.

![Screenshot from 2023-10-12 17-24-25](https://github.com/kuzudb/kuzu/assets/21187588/933047e9-fb93-4ee0-abec-decec44cda92)
